components/net/fetch/methods.rs

```diff
@@ -569,17 +569,17 @@ pub enum RangeRequestBounds {
 }
 
 impl RangeRequestBounds {
-    pub fn get_final(&self, len: Option<u64>) -> Result<RelativePos, ()> {
+    pub fn get_final(&self, len: Option<u64>) -> Option<RelativePos> {
         match self {
             RangeRequestBounds::Final(pos) => {
                 if let Some(len) = len {
                     if pos.start <= len as i64 {
-                        return Ok(pos.clone());
+                        return Some(pos.clone());
                     }
                 }
-                Err(())
+                None
```

Comment:
For instance this could be:

```suggestion
                Err("Tried to process RangeRequestBounds::Final without len")
```

or something along those lines.